src/Microsoft.OData.Core/Batch/ODataBatchReader.cs

```diff
@@ -374,6 +371,82 @@ protected virtual string GetCurrentGroupIdImplementation()
+        /// in a batch request.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains an <see cref="ODataBatchOperationRequestMessage"/>
+        /// that can be used to read the content of the batch request operation from.
+        /// </returns>
+        protected virtual Task<ODataBatchOperationRequestMessage> CreateOperationRequestMessageImplementationAsync()
+        {
+            return TaskUtils.GetTaskForSynchronousOperation(this.CreateOperationRequestMessageImplementation);
+        }
+
+        /// <summary>
+        /// Returns the cached <see cref="ODataBatchOperationResponseMessage"/> for reading the content of an operation
+        /// in a batch response.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains an <see cref="ODataBatchOperationResponseMessage"/>
+        /// that can be used to read the content of the batch response operation from.
+        /// </returns>
+        protected virtual Task<ODataBatchOperationResponseMessage> CreateOperationResponseMessageImplementationAsync()
+        {
+            return TaskUtils.GetTaskForSynchronousOperation(this.CreateOperationResponseMessageImplementation);
+        }
+
+        /// <summary>
+        /// Asynchronous implementation of the reader logic when in state 'Start'.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the batch reader state after the read.
+        /// </returns>
+        protected virtual Task<ODataBatchReaderState> ReadAtStartImplementationAsync()
+        {
+            return TaskUtils.GetTaskForSynchronousOperation(this.ReadAtStartImplementation);
+        }
+
+        /// <summary>
+        /// Asynchronous implementation of the reader logic when in state 'Operation'.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the batch reader state after the read.
+        /// </returns>
+        protected virtual Task<ODataBatchReaderState> ReadAtOperationImplementationAsync()
+        {
+            return TaskUtils.GetTaskForSynchronousOperation(this.ReadAtOperationImplementation);
+        }
+
+        /// <summary>
+        /// Asynchronous implementation of the reader logic when in state 'ChangesetStart'.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the batch reader state after the read.
+        /// </returns>
+        protected virtual Task<ODataBatchReaderState> ReadAtChangesetStartImplementationAsync()

```

Comment:
Why are we wrapping a sync operation in an async method?

Reply:
@KenitoInc We're doing this to avoid a breaking change. Several classes inherit from `ODataBatchReader`. Since all those classes at the moment implement the sync API, the async wrapper over the async is a better way than throwing a `NotImplementedException`. Since the async methods are declared as virtual, they are overridden in the derived classes and this stubbed implementation never gets executed